### ISSUE TITLE
ios login open a new url

### ISSUE THREAD

**Issue by chenkuangkuang** [line/line-sdk-ios-swift#196]
hello. My app use  ["react-native-line"](https://github.com/xmartlabs/react-native-line) to archive line login feature. It works well In android. But In IOS, after show authorize page, it will open another authorize page which url is ""access.line.me"(in line app).

![image](https://github.com/line/line-sdk-ios-swift/assets/20104135/6e1ae424-708b-45e6-a2be-0f8d6692baa9)

I have asked the develper, and they replied that this is default of Line.

So I create a issue here, and want to know is this default ?

if not , how can I skip the page?

**Comment by onevcat** [comment]
Hi, @chenkuangkuang 

TLDR: Set up a universal link for your app, register it on the developer console and use it to initialize the LINE Login.

This is the default behavior when using the `line://` scheme and allow your user to confirm the app they are going to. Remember that it is not safe to use a customize URL scheme to open the auth URL containing the auth `code`, so it is a way to improve the safety. (There are multiple other methods to ensure the login process is secured, but this provides a visual way to protect the auth code from the beginning.)

To "skip" the screen, you can instead of using the customize URL scheme, set up and switch to use the universal link to open your app. Please refer to the related guide: https://developers.line.biz/en/docs/line-login-sdks/ios-sdk/swift/universal-links-support/

**Comment by onevcat** [comment]
Hi,

Sorry for the late reply.

Your code can work. However, the more accurate way should be checking the result of LINE SDK: If the URL can be handled by LINE SDK, it returns a true bool value and you can stop. Otherwise, you can pass it to your other handler for future processing.

```
- (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull NSUserActivity *)userActivity
 restorationHandler:(nonnull void (^)(NSArray<id<UIUserActivityRestoring>> * _Nullable))restorationHandler
{

    BOOL handledLine = [LineLogin application:application continue:userActivity restorationHandler:restorationHandler];
    if (handledLine) {
        return YES;
    }
  
    if ([userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb]){
      return [RCTLinkingManager application:application
                  continueUserActivity:userActivity
                    restorationHandler:restorationHandler];
    }

    // And your other handlers, if there is one.
}
```

I would suggest you put your `LineLogin` code at the top of the related methods, to prevent the auth `code` being pass to another third-party providers (that would be a risk of leaking the code).

**Comment by onevcat** [comment]
I don't think it can cause any performance issue. Inside LINE SDK, it is doing the URL check in very early stage and won't do any additional work besides of matching the URL. If it is not a URL can be recognized by our code, it returns `false` as soon as possible. 

I believe unless you are doing more than thousands of app opening or user activity continuing (which is ridiculers itself), passing the URL to LINE SDK won't cause any issue.
